src/main/java/com/brandwatch/internship/fetchandstoretweets/services/FetchAndStoreService.java

```diff
@@ -0,0 +1,67 @@
+package com.brandwatch.internship.fetchandstoretweets.services;
+
+import com.brandwatch.internship.fetchandstoretweets.entities.Mention;
+import com.brandwatch.internship.fetchandstoretweets.entities.Query;
+import com.brandwatch.internship.fetchandstoretweets.repositories.MentionsRepository;
+import com.brandwatch.internship.fetchandstoretweets.repositories.QueryRepository;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.social.twitter.api.SearchResults;
+import org.springframework.social.twitter.api.Tweet;
+import org.springframework.social.twitter.api.Twitter;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+@Service
+public class FetchAndStoreService {
+
+    private Twitter twitter;
+
+    private MentionsRepository mentionsRepository;
+    private QueryRepository queryRepository;
+
+    private int counter;
+    private String[] searchStrings = {"#maven", "#spring", "#intellij_idea", "#kafka"};
+
+    @Autowired
+    public FetchAndStoreService(Twitter twitter, MentionsRepository mentionsRepository,
+                                QueryRepository queryRepository) {
+
+        this.twitter = twitter;
+        this.mentionsRepository = mentionsRepository;
+        this.queryRepository = queryRepository;
+
+        this.mentionsRepository.deleteAll();
+        this.queryRepository.deleteAll();
+
+    }
+
+    @Scheduled(fixedRate = 5000)
+    public void run() {
+
+        Query query = new Query(counter + 1, searchStrings[counter++ % 4]);
```

Comment:
You need to get the id using the database. Anything that's happening in the app will get lost when you restart (like your `AtomicLong` solution before, or now the counter)

Reply:
Yeah, I will solve this one.

Reply:
`counter++ % 4` is an expression with a magic number, which is difficult to understand. It is better to extract it to a separate method `getNextQuery` and use constant `searchStrings.length`

Reply:
u r right. It should have been `searchStrings.length` and in a separate method to improve readability. I will change it completely anyway to make it depend on the DB.